modules/apps/frontend-js/frontend-js-spa-web/test/app/App.ts

```diff
@@ -2048,46 +2276,56 @@ describe('App', function () {
 				return 'title';
 			}
 		}
-		this.app = new App();
-		this.app.addRoutes(new Route('/path', TitledScreen));
-		this.app.navigate('/path').then(() => {
+		app = new App();
+		app.addRoutes(new Route('/path', TitledScreen));
+		app.navigate('/path').then(() => {
 			expect(document.title).toBe('title');
 			done();
 		});
 	});
 
 	it('sets Liferay.SPA.__capturedFormElement__ to null after navigate', (done) => {
-		this.app = new App();
-		this.app.addRoutes(new Route('/path', Screen));
-		this.app.navigate('/path').then(() => {
+		app = new App();
+
+		app.addRoutes(new Route('/path', Screen));
+
+		app.navigate('/path').then(() => {
+			if (!Liferay.SPA) {
+				return;
+			}
+
 			expect(Liferay.SPA.__capturedFormElement__).toBeNull();
 			done();
 		});
 	});
 
 	it.skip('cancels nested promises on canceled navigate', (done) => {
-		this.app = new App();
-		this.app.addRoutes(new Route('/path', HtmlScreen));
-		this.app
-			.navigate('/path')
+		app = new App();
+		app.addRoutes(new Route('/path', HtmlScreen));
+		app.navigate('/path')
 			.then(() => done.fail())
 			.catch(() => {
 				expect(fetch.mock.calls.length).toBe(0);
 				done();
 			})
+
+			// @ts-ignore
+
 			.cancel();
 	});
 
 	it.skip('cancels nested promises on canceled prefetch', (done) => {
-		this.app = new App();
-		this.app.addRoutes(new Route('/path', HtmlScreen));
-		this.app
-			.prefetch('/path')
+		app = new App();
+		app.addRoutes(new Route('/path', HtmlScreen));
+		app.prefetch('/path')
 			.then(() => done.fail())
 			.catch(() => {
 				expect(fetch.mock.calls.length).toBe(0);
```

Comment:
`Property 'mock' does not exist on type '(input: RequestInfo, init?: RequestInit | undefined) => Promise<Response>'`

🤷  haven't looked much into this one, but might be a limitation of our current testing architecture